Update draft-poidt-ccamp-actn-poi-pluggable.md

ggalimba56 opened the pull request:
commit test

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

test 2

Commit:
added the modification upon comments

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -218,16 +227,15 @@ o--/---o               o---\-|-------|--/---o               o---\--o
-P.N. = Packet Node (ROADM)
-O.N. = Optical DWDM Node
+P.N. = Packet/Optical Node (IPoDWDM router)
+O.N. = Optical Switching DWDM Node (ROADM)
```
The packet node shoudln't be a ROADM right?

ggalimba56 replied:
Correct, It will be fixed

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -237,23 +245,181 @@ Px = DWDM (coherent pluggable) Router ports
+ ## Option 2 - Single SBI management of IPoWDM routers
+
+   The architecture related to option 2 is more meeting a classical
+   ACTN approach seen also in {{!RFC8453}} and 
```
Yes and no. The hierarchical approach is still there, but in option 1 we have the optical PNC that connects to the packet nodes. It's not a classical ACTN architecture.

sergiobelotti replied:
```suggestion
   The architecture related to option 2 is more meeting a canonical  ACTN approach, i.e. any PNCs only manage resources related to his own administrative domain,  like in {{!RFC8453}} and 
```

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -311,43 +479,44 @@ Px = DWDM (coherent pluggable) Router ports
    parameters are read by the P-PNC from the DWDM pluggable and shared
    with MDSC to give the visibility of the pluggable characteristics.
-   MDSC can use the info to understand the client capabolity and, again,
+   MDSC can use the info to understand the client capability and, again,
```
yes i guess so.

ggalimba56 replied:
change "client" to "pluggable capabilities"

ggalimba56 commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -147,42 +147,51 @@ informative:
+   multilayer network use cases where the DWDM optical transmission
+   function is located within the packet domain equipment instead of 
+   being part of the Optical domain {{my_figure-1}}.
+   This means that an IP/MPLS capable device becomes a multi-layer
```
OK !  accepted

ggalimba56 commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -147,42 +147,51 @@ informative:
 
-   ZR and ZR+ (and also CFP2-DCO) deployment in routers has already
-   started and are expanding significantly.  The way the DWDM pluggable
+   The pluggable DCO deployment in routers has already
```
let's use coherent interface instead of DCO

ggalimba56 commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -268,13 +434,15 @@ Px = DWDM (coherent pluggable) Router ports
-   PDL, FEC, etc.  For more details refer to draft-ietf-ccamp-dwdm-
-   if-param-yang.  The above parameters may not be directly retrieved
+   PDL, FEC, etc.  For more details refer to 
+   {{!I-D.draft-ietf-ccamp-dwdm-if-param-yang}}. 
```
OK !

ggalimba56 commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -311,43 +479,44 @@ Px = DWDM (coherent pluggable) Router ports
+      (between transceiver and ROADM) loss comparing it with the ROADM
+      port received power.
 
    operational-mode
```
Yes, need to clarify better.

ggalimba56 commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -311,43 +479,44 @@ Px = DWDM (coherent pluggable) Router ports
    DWDM Pluggable.  The pluggable provisioning will be done by the
    P-PNC.  For more details on the optical interface parameters see:
-   {{!I-D.ietf-ccamp-dwdm-if-param-yang}}.
+   {{!I-D.draft-ietf-ccamp-dwdm-if-param-yang}}.
```
OK !

ggalimba56 commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -404,13 +573,13 @@ DCO-link = Pluggable connection (OTSi connection)
 MC-link = Media Channel link (MC optical circuit)
```
Yes, we need to clarify and provide the model of the OTSi vs. MC

DanieleIETF replied:
v02 - Add that both options are applicable here.

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

Version addressing the changes in the .doc file (daniele, oscar, gabriele, phil and sergio)

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

Jeff's comments addressed.
Still to be fixed:
- Add MANTRA reference
- Substitute P.N. in the figures with IP/MPLS nodes
- Close on definition of OTSi/Media Channel/Network Medial Channel
- Fix reference to Optical Imp Topo.

ggalimba56 committed:
Add files via upload

text file generated from .md

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

fixed picture 3

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

modified acknowledgments

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

added Mantra and impairments-topology draft references

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

Test mantra reference

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

mantra test

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

Added Oscar as co-author

ggalimba56 committed:
Add files via upload

.txt file with last modifications based on team comments.

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

Co-authored-by: sergio belotti <sergio.belotti@alcatel-lucent.com>

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -311,43 +479,44 @@ Px = DWDM (coherent pluggable) Router ports
+      only one operational-mode to be set, together with central 
+      frequency and TX power, in the pluggable though MDSC and P-PNC.
+
+   The above optical parameters are related to the Edge Node Transceiver
```
Offline

DanieleIETF replied:
Will be fixed in 02

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -423,8 +592,8 @@ MC-link = Media Channel link (MC optical circuit)
+      {{!I-D.draft-ietf-teas-actn-poi-applicability}}
 
    End to End Packet service provisioning / deletion
 
```
used router everywhere

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -436,8 +605,8 @@ MC-link = Media Channel link (MC optical circuit)
-      Connectivity between two Routers, the IP link will be set up later
-      using this optical tunnel.
+      Connectivity between two Routers, the IP link will be set up
+      later using this optical tunnel.
```
v02

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -482,12 +657,16 @@ MC-link = Media Channel link (MC optical circuit)
             - Manufacturing data
 
-   It would be also possible to auto-discover the inter-layer (inter-
+   It would be also possible to auto-discover the inter-domain (inter-
```
v02

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -147,42 +166,51 @@ informative:
+   Coherent pluggable digital coherent optics, such as ZR 
+   {{OIF-400ZR-01-0}} and ZR+ {{Open_ZR-Plus_MSA}}, are enabling new 
+   multilayer network use cases where the DWDM optical transmission
+   function is located within the packet domain equipment instead of 
```
We need to be very careful with the term "domain". From my viewpoint, the "packet domain" is concerned with packet processing and not photonics and hence an equipment that has photonics is NOT in the packet domain, it straddles the two domains. A device straddling multiple domains is well recognized as there are many cases where a device spreads across multiple domains (admin, protection, technology etc.).

DanieleIETF replied:
v02

nigel-r-davis replied:
Whilst I understand that there is significant work in adjustment to the notion of domains, I think the current rationale is flawed. I would minimally want to see some acknowledgement that the photonics is not really in the packet domain and that some refinement is required here.

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -147,42 +166,51 @@ informative:
    any standard and it is becoming an urgent matter to cover for Service
-   Providers.  Full end-to-end management solution of these DWDM
-   coherent pluggable optics, leveraging on ACTN hierarchical
+   Providers.  A full end-to-end management solution of these pluggable
```
I suggest that this be reworded… "A full end-to-end management solution of these pluggable digital coherent optics is becoming critical to allow a wider deployment beyond simple point-to-point high-capacity link scenarios between two IP/ MPLS routers. It is proposed that this solution leverages and builds on the ACTN hierarchical architecture."

nigel-r-davis replied:
OK. I think the paragraph is overstating the ACTN usage, but this is not a critical point.

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -147,42 +166,51 @@ informative:
+                            +-+-++-+-+-+
+                              | |  | |
+               +--------------+ |  | +--------------+
+   MPI interf. |           +----+  +----+           |
```
Considering my comment on Domains earlier, the node names in this figure need some adjustment. I propose simple numbering of nodes with no mention of technology/domain.

DanieleIETF replied:
v02

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -237,29 +264,184 @@ Px = DWDM (coherent pluggable) Router ports
 # Reference architecture and network scenario
```
This is clearly a reflection of solutions today and is not looking forward to where the solutions are likely to eventually land (single control fabric). That's OK so long as the intention of this document is to not push industry advancement.

It is expected that the single control fabric will be an assembly of fine-grained components from various vendors providing a holistic treatment of the entire network not fragmented by traditional artificial boundaries. It is also likely that there will be specialist components focused on key aspects such as photonic technology where these components have direct access to the relevant properties of the device (so a component dealing with photonic technology would access the device "directly" for monitoring and control of photonic parameters.

DanieleIETF replied:
v02

nigel-r-davis replied:
OK. V02 is fine here. I would like to see more thinking wrt the target in V02.

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -237,29 +264,184 @@ Px = DWDM (coherent pluggable) Router ports
+   The peculiarity of this option consists of the fact that both the 
+   packet SDN controller (P-PNC) and the optical SDN controller (O-PNC)
+   have access to the coherent pluggable optics on the routers. 
+   The P-PNC is the only entity allowed to configure them, while the
```
It would seem more reasonable that the O-PNC would have read/write access to the photonics on the plug and the P-PNC would not. The devices straddles the two domains.

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -237,29 +264,184 @@ Px = DWDM (coherent pluggable) Router ports
+   packet SDN controller (P-PNC) and the optical SDN controller (O-PNC)
+   have access to the coherent pluggable optics on the routers. 
+   The P-PNC is the only entity allowed to configure them, while the
+   O-PNC is granted with read-only permissions to avoid database 
```
Database inconsistencies should not occur if the appropriate responsibility demarcation is drawn and config locking performed.

There are examples of successful deployment of devices with several separate controllers, indeed some are required by particular operational practices.

DanieleIETF replied:
v02

nigel-r-davis replied:
I would like to discuss this.

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -237,29 +264,184 @@ Px = DWDM (coherent pluggable) Router ports
+       counters.
+     - Fault notification, received asynchronous alarm notifications.
+
+   The consequence of this split of roles is that the P-PNC exposes
```
Agreed, there would be these consequences. This could be an architectural choice, but it is clearly only one possible approach and it also appears to not be the best approach

nigel-r-davis commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -237,29 +264,184 @@ Px = DWDM (coherent pluggable) Router ports
+   and service provisioning for OLS media channel services.
+
+
+ ## Option 2 - Single SBI management of IPoWDM routers
```
I propose, that minimally an option 3 be added that recognizes the domain boundary in the device and explains that the optical controller controls the photonics of the plug. I would be happy to write up this option (in a consistent form with the other options) and provide the use case details.

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.txt:
```diff
@@ -0,0 +1,1456 @@
+
+   The first operation executed by the P-PNC and O-PNC is to discover
+   the network topology and share it with the MDSC via the MPI.  The
+   PNCs will discover and share also the inter-layer links (or
```
The section above says:


It would be also possible to auto-discover the inter-domain
   links between DWDM coherent pluggables and ROADM ports by
   checking the input/output power levels (and probably switching on/off
   the lasers of the pluggables).  This would require the help of MDSC,
   O-PNC and P-PNC.  The same method could be used to verify the
   provisioned connectivity. For further study in this draft.

What is missing? Maybe describe better the interaction between MDSC, O-PNC and P-PNC ?

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -125,18 +135,27 @@ normative:
+   target:  https://cdn.brandfolder.io/D8DI15S7/at/n85t9h48bqtkhm9k7tqbs9fv/TIP_OOPT_MANTRA_IP_over_DWDM_Whitepaper_-_Final_Version3.pdf
+
 
 --- abstract
```
v02

nigel-r-davis replied:
I think the draft is weakened without this acknowledgement.

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -147,42 +166,51 @@ informative:
+   controlling its IP/MPLS domain, which can be either an Autonomous 
+   System (AS), {{!RFC1930}}, or an IGP area within the same operator 
+   network.  Each Optical PNC (O-PNC) in the below topology is 
+   responsible for controlling its own Optical Domain.
```
v02

DanieleIETF commented on draft-poidt-ccamp-actn-poi-pluggable.md:
```diff
@@ -237,29 +264,184 @@ Px = DWDM (coherent pluggable) Router ports
+{: #my_figure-2 title="Dual SBI management Scenario"}
+
+
+   The peculiarity of this option consists of the fact that both the 
```
v02

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

Updated the document version to -01 and reformatted the text

ggalimba56 committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

fixed some indentations

ggalimba56 committed:
Add files via upload

draft version submitted to IETF

ggalimba56 committed:
Add files via upload

draft version submitted to IETF

DanieleIETF committed:
Update draft-poidt-ccamp-actn-poi-pluggable.md

Added EU season project ACK and fixed section 2.2

oscargdd committed:
changed to -latest

oscargdd committed:
fixed lint errors

oscargdd committed:
Fixed editorial issues
